commands/proxy/proxy-set.ts

```diff
@@ -22,22 +23,30 @@ export class ProxySetCommand extends ProxyCommandBase {
 	}
 
 	public async execute(args: string[]): Promise<void> {
-		let hostname = args[0],
-			port = args[1],
-			username = args[2],
-			password = args[3];
+		let urlString = args[0],
+			username = args[1],
+			password = args[2];
+
+		const noUrl = !urlString;
+		if (noUrl) {
+			if (!isInteractive()) {
+				this.$errors.fail("Console is not interactive - you need to supply all command parameters.");
+			} else {
+				urlString = await this.$prompter.getString("Url", { allowEmpty: false });
+			}
+		}
 
-		const noHostName = !hostname;
+		const urlObj = parse(urlString);
+		let port = urlObj.port && +urlObj.port || HttpProtocolToPort[urlObj.protocol];
 		const noPort = !port || !this.isValidPort(port);
+		const authCredentials = this.getCredentialsFromAuth(urlObj.auth || "");
+		username = username || authCredentials.username;
+		password = password || authCredentials.password;
 
-		if (!isInteractive() && (noHostName || noPort || this.isPasswordRequired(username, password))) {
+		if (!isInteractive() && (noPort || this.isPasswordRequired(username, password))) {
```

Comment:
I know I've written this part, but now I see an incorrect behavior here - In case I've specified port, but it's not valid and the terminal is not interactive, we'll receive error "Console is not interactive - you need to supply all command parameters."
Can you fix this by providing correct message in such case?